src/core/mount.c

```diff
@@ -1197,19 +1197,19 @@ static void mount_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         case MOUNT_MOUNTING_SIGKILL:
         case MOUNT_MOUNTING_SIGTERM:
 
-                if (f == MOUNT_SUCCESS)
-                        mount_enter_mounted(m, f);
+                if (m->result == MOUNT_SUCCESS)
+                        mount_enter_mounted(m, m->result);
                 else if (m->from_proc_self_mountinfo)
-                        mount_enter_mounted(m, f);
+                        mount_enter_mounted(m, m->result);
                 else
-                        mount_enter_dead(m, f);
+                        mount_enter_dead(m, m->result);
```

Comment:
This doesn't actually change anything because ``mount_enter_mounted()`` and ``mount_enter_dead()`` will check ``m->result`` again.